File: src/main/java/cn/cloudself/model/CarEntity.java
package cn.cloudself.model;

import javax.persistence.*;

/**
 * @author HerbLuo
 * @version 1.0.0.d
 */
@Entity
@Table(name = "car", schema = "shop")
public class CarEntity {
    private int id;
    private int userId;
    private ItemEntity item;

    @Id
    @Column(name = "id", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Basic
    @Column(name = "user_id", nullable = false)
    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        CarEntity carEntity = (CarEntity) o;

        if (id != carEntity.id) return false;
        if (userId != carEntity.userId) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = id;
        result = 31 * result + userId;
        return result;
    }

    @ManyToOne
    @JoinColumn(name = "item_id", referencedColumnName = "id", nullable = false)
    public ItemEntity getItem() {
        return item;
    }

    public void setItem(ItemEntity item) {
        this.item = item;
    }

    public CarEntity() {
    }

    public CarEntity(int userId, ItemEntity item) {
        this.userId = userId;
        this.item = item;
    }

    @Override
    public String toString() {
        return "CarEntity{" +
                "id=" + id +
                ", userId=" + userId +
                ", item=" + item +
                '}';
    }
}

File: src/main/java/cn/cloudself/model/CarEntity.kt
package cn.cloudself.model

import javax.persistence.*

/**
 * @author HerbLuo
 * @version 1.0.0.d
 */
@Entity
@Table(name = "car", schema = "shop")
data class CarEntity(

        @get:Id
        @get:Column(name = "id", nullable = false)
        @get:GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Int = 0,

        @get:Basic
        @get:Column(name = "user_id", nullable = false)
        var userId: Int = 0,

        @Suppress("MemberVisibilityCanPrivate")
        @get:ManyToOne
        @get:JoinColumn(name = "item_id", referencedColumnName = "id", nullable = false)
        var item: ItemEntity? = null

) {
    constructor(id: Int, item: ItemEntity?) : this() {
        this.id = id
        this.item = item
    }
}
